tsl/src/nodes/decompress_chunk/planner.c

```diff
@@ -891,8 +913,14 @@ find_vectorized_quals(DecompressionMapContext *context, DecompressChunkPath *pat
 		 */
 		Node *transformed_comparison =
 			(Node *) ts_transform_cross_datatype_comparison((Expr *) source_qual);
-
-		Node *vectorized_qual = make_vectorized_qual(context, path, transformed_comparison);
+		VectorQualInfoDecompressChunk vqidc = {
+			.vqinfo = {
+				.vector_attrs = build_vector_attrs_array(context->uncompressed_attno_info, path->info),
```

Comment:
Nit: This this variable goes out of scope and you do not copy or keep a reference to this field in `vector_qual_make` you might want to free this memory before exiting the function. It is released at the end of the transaction anyway, so there is no memory leak, but since you know that the memory is not used it can be released.